metrics/testing/pom.xml

```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Copyright (c) 2023 Oracle and/or its affiliates.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.helidon.metrics</groupId>
+        <artifactId>helidon-metrics-project</artifactId>
+        <version>4.0.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <artifactId>helidon-metrics-testing</artifactId>
```

Comment:
We might want not to move the tests under the provider submodule.

1. Although the test module is related to providers it is not itself a provider. It might seem a little odd.
2. It's nice to add to the `providers/pom.xml` file the dependency on the tests artifact. Every provider implementation's pom should specify the providers project as is parent, so each provider will inherit and run the tests automatically.

For the moment I have renamed the module to `helidon-metrics-provider-tests` but left it at the same place in the hierarchy. 

We can revisit this later if we need to.

Reply:
That surefire set-up allows the child submodules to automatically run the tests declared in that referenced module.

Reply:
renaming to `helidon-metrics-provider-tests` is OK, thanks.
I would like to have `testing` reserved for modules for our users (that help them test applications), and `tests` for actual tests of our own.

Reply:
done